2018/004-image-update-caasp-v4.md

```diff
@@ -0,0 +1,660 @@
+Image tagging and update strategy for CaaSP 4.0
+
+| Field  | Value      |
+|:-------|:-----------|
+| Status | Draft      |
+| Date   | YYYY-MM-DD |
+
+## Introduction
+
+>This section targets end users, operators, PM, SEs and anyone else that might
+> need a quick explanation of your proposed change.
+
+CaaSP has always been running some of its components using containers. Container
+images in the past have always been delivered via RPM packages. On
+CaaSP all packages are managed by the `transactional-update` tool. This
+technology requires nodes to be rebooted every time a new package is installed.
+The update mechanism of CaaSP, both for RPMs and for container images, has been
+designed around that.
+
+Starting from v4 all the container images are going to be distributed using the
+SUSE registry. That means RPM packages are not going to be used to install and
+update containers.
+
+Due to this change we have to revisit the update story for container images.
+
+
+
+Before we proceed into the details of this RFC, it's important to outline how
+our images are being tagged.
+
+All our images are built inside of the build service and have at least two tags
+associated with them:
+
+  * **short:** `<app-version>` this is something like: `2.6.1`
+  * **intermediate:** `<app-version>-<image-version>` this is something like:
+    `2.6.1-1`. This value is hard-coded into the KIWI image definition; image
+    owners are going to bump it manually.
+  * **long:** `<app-version>-<image-version>-<buildID>`. The `buildID` is a
+    value automatically generated by the Open Build Service. This tag would be
+    something like `2.6.1-1.100`. Note well the `buildID` changes after each
+    build and is specific to the project where the image is being built.
+    To be more explicit: the same image built inside of the `Devel:CaaSP`
+    project will have different numbers compared to very same one built inside
+    of `SUSE:CaaSP`.
+
+### Problem description
+
+> Why do we want this change and what problem are we trying to address?
+
+Starting with v4 the need to reboot a node to get newer container images is no
```

Comment:
Starting with v4 it is no longer required to reboot a node to get newer container images: